kratos/includes/geometry_container.h

```diff
@@ -0,0 +1,407 @@
+    ///@{
+
+    bool HasGeometry(IndexType GeometryId) const
+    {
+        return (mpGeometries->find(GeometryId) != mpGeometries->end());
+    }
+
+    bool HasGeometry(std::string GeometryName) const
+    {
+        auto hash_index = TGeometryType::GenerateId(GeometryName);
+
+        return (mpGeometries->find(hash_index) != mpGeometries->end());
+    }
+
+    ///@}
+    ///@name Iterators
+    ///@{
+
+    GeometryIterator GeometriesBegin()
+    {
+        return mpGeometries->begin();
+    }
+
+    GeometryConstantIterator GeometriesBegin() const
+    {
+        return mpGeometries->begin();
+    }
+
+    GeometryIterator GeometriesEnd()
+    {
+        return mpGeometries->end();
+    }
+
+    GeometryConstantIterator GeometriesEnd() const
+    {
+        return mpGeometries->end();
+    }
+
+    ///@}
+    ///@name Container Functions
+    ///@{
+
+    GeometriesContainerType& Geometries()
+    {
+        return *mpGeometries;
+    }
+
+    const GeometriesContainerType& Geometries() const
+    {
+        return *mpGeometries;
+    }
+
+    typename GeometriesContainerType::Pointer pGeometries()
+    {
+        return mpGeometries;
+    }
+
+    void SetGeometries(typename GeometriesContainerType::Pointer pOtherGeometries)
+    {
+        mpGeometries = pOtherGeometries;
+    }
+
+    //typename GeometriesContainerType::ContainerType& GeometriesArray()
+    //{
+    //    return mpGeometries->GetContainer();
+    //}
+
+    ///@}
+    ///@name Container Functions
+    ///@{
+
+    //GeometriesContainerType& Geometries(GeometryData::KratosGeometryType GeometryType)
+    //{
+    //    return *mpGeometries->where(...);
+    //}
+
+    ///@}
+    ///@name Container Functions
+    ///@{
+
+    //GeometriesContainerType& Geometries(GeometryData::KratosGeometryType GeometryType)
+    //{
+    //    return *mpGeometries->where(...);
+    //}
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Return information
+    std::string Info() const override
+    {
+        return "GeometryContainer";
+    }
+
+    /// Print information about this object
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+        rOStream << "Number of Geometries: " << mpGeometries->size() << std::endl;
+    }
+
+    /// Print information about this object
+    virtual void PrintInfo(std::ostream& rOStream, std::string const& PrefixString) const
+    {
+        rOStream << PrefixString << Info();
+    }
+
+    /// Print object's data
+    virtual void PrintData(std::ostream& rOStream, std::string const& PrefixString ) const
+    {
+        rOStream << PrefixString << "Number of Geometries: " << mpGeometries->size() << std::endl;
+    }
+
+    ///@}
+
+private:
+    ///@name Member Variables
+    ///@{
+
+    /// Geometry Container
+    typename GeometriesContainerType::Pointer mpGeometries;
```

Comment:
should be a pointer? asking also the others

Reply:
I am open for discussion. This is the way it was done in the mesh.

Reply:
It is now a member, as it should be...